src/Canvas.tsx

```diff
@@ -319,13 +321,23 @@ const InternalCanvas: FC<CanvasProps & { ref?: Ref<CanvasRef> }> = forwardRef(
     );
 
     useEffect(() => {
-      if (dragNodeData.children) {
+      if (dragNodeData) {
```

Comment:
Putting elements in state doesn't feel like a good idea.

Reply:
There needs to be a way to identify and dynamically generate the JSX element based on the `dragNode` callback. Do you have any recommendations on a better approach?

Reply:
I think I have a good solution. Changes incoming